specs/account_spec.rb

```diff
@@ -136,36 +138,56 @@
   end
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Wave 2" do
+describe "Wave 2" do
   describe "Account.all" do
     it "Returns an array of all accounts" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - Account.all returns an array
-      #   - Everything in the array is an Account
-      #   - The number of accounts is correct
-      #   - The ID and balance of the first and last
-      #       accounts match what's in the CSV file
-      # Feel free to split this into multiple tests if needed
+      Bank::Account.all.must_be_instance_of Array
+
+      Bank::Account.all.each do |account|
+        account.must_be_instance_of Bank::Account
+      end
+    end
+
+    it "Contains the correct amount of accounts" do
+      Bank::Account.all.length.must_equal 12
+    end
+
+    it "Matches the ID and balance of the first and last accounts" do
+      Bank::Account.all.first.id.must_equal 1212
```

Comment:
This looks good!

Something to think about: every time you call `Bank::Account.all`, Ruby has to go and re-read the entire CSV file. This isn't such a big deal when our file is only 12 lines, but in a real world scenario you might be working with a very large file, which would take a long time to load. In that case it would be wise to save the results of `Bank::Account.all` to a variable and make your checks on it. Then you only have to load the file once.